lib/pure/terminal.nim

```diff
@@ -19,22 +19,33 @@
 import macros
 import strformat
 from strutils import toLowerAscii
-import colors
+import colors, tables
 
-const
-  hasThreadSupport = compileOption("threads")
+when defined(windows):
+  import winlean
 
-when not hasThreadSupport:
-  import tables
-  var
-    colorsFGCache = initTable[Color, string]()
-    colorsBGCache = initTable[Color, string]()
-    styleCache = initTable[int, string]()
+type
+  PTerminal = ref object
+    colorsFGCache: Table[Color, string]
```

Comment:
If you don't mind, remove these caches. If terminal output is one's bottleneck, one can always use the escape codes directly.